sdk/inc/azure/core/internal/az_http_internal.h

```diff
@@ -193,19 +198,39 @@ AZ_NODISCARD az_result az_http_request_init(
  * For instance, if url in request is `http://example.net?qp=1` and this function is called with
  * path equals to `test`, then request url will be updated to `http://example.net/test?qp=1`.
  *
+ * @remarks if \p is_path_url_encoded is set to false, before appending the path, it would be
+ * url-encoded.
  *
- * @param ref_request http request builder reference
- * @param path span to a path to be appended into url
- * @return AZ_NODISCARD az_http_request_append_path
+ *
+ * @param[out] ref_request http request reference
+ * @param[in] path span to a path to be appended into url
+ * @param[in] \p is_path_url_encoded boolean value that defines if the value to be appended is
+ * url-encoded or not.
+ * @return
+ *   - *`AZ_OK`* success.
+ *   - *`AZ_ERROR_INSUFFICIENT_SPAN_SIZE`* the `URL` would grow past the `max_url_size`, should
+ * the parameter get set.
+ *   - *`AZ_ERROR_ARG`*
+ *     - `ref_request` is _NULL_.
  */
-AZ_NODISCARD az_result az_http_request_append_path(az_http_request* ref_request, az_span path);
+AZ_NODISCARD az_result
+az_http_request_append_path(az_http_request* ref_request, az_span path, bool is_path_url_encoded);
```

Comment:
url_encoded is a better name.

Reply:
don't you think it is not as descriptive ?  I was trying to make it as much self-descriptive as possible

Reply:
We could also do `is_encoded`.

Otherwise, `url_encode` is nice. We should append the name of the url that we will encode: `url_encode_path`, similar to @JeffreyRichter's suggestion below for `url_encode_name` and `url_encode_value`

Reply:
In any case, this is an internal method so parameter names are not as important.

Reply:
These are the 2 options I considered for the boolean flag:
-  `url_encode` :  It means that the API is asking whoever is using the API if they want to url-encode the values that they are passing. It means that anyone consuming this API needs to know if they want to url-encode values or not. Boolean set to `true` here would url-encode values passed

-  `is_url_encoded`:  Means that anyone consuming the API doesn't need to decide if they want to url-encode or not. All they need to know is if the values are url-encoded already. A boolean value `false` here would url-encode values. (this is the implemented behavior)

The first option can lead someone to think "why do I want to url-encode?"  and just pass `false`. It is basically asking about the future behavior.
The second option asks information about the data that is passed.  API controls the future behavior based on the input properties and not on user choice.

I can change the name, but I strongly believe that second option is a better way to go

Reply:
> I can change the name, but I strongly believe that second option is a better way to go

Sounds good to me.